ipld/cid/src/lib.rs

```diff
@@ -122,10 +122,23 @@ impl<'de> de::Deserialize<'de> for Cid {
             Some(CBOR_TAG_CID) | None => {
                 let mut bz = tagged.value.into_vec();
 
-                if bz.first() == Some(&MULTIBASE_IDENTITY) {
-                    bz.remove(0);
+                if bz.len() == 0 {
+                    return Err(de::Error::custom("undefined cid"));
                 }
 
+                if bz.len() < 2 {
+                    return Err(de::Error::custom(
+                        "cbor serialized CIDs must have at least two bytes",
+                    ));
+                }
+
+                if bz.first() != Some(&MULTIBASE_IDENTITY) {
```

Comment:
This looks good. I'd like to see unit tests covering the different behaviors.

Reply:
@lemmih pr is updated, pls check

Reply:
Aligning with Lotus would break these tests: https://github.com/ChainSafe/forest/blob/main/ipld/tests/cbor_test.rs#L20 because 1 at index 0 is no longer allowed. Not sure if the tests should be updated.

Reply:
Yes, those tests have to be updated.